content/master/concepts/compositions.md

```diff
@@ -0,0 +1,993 @@
+create a new AWS virtual network. This includes a 
+{{<hover label="xnetwork" line="8">}}VPC{{</hover>}},
+{{<hover label="xnetwork" line="13">}}InternetGateway{{</hover>}} and
+{{<hover label="xnetwork" line="18">}}Subnet{{</hover>}}.
+
+```yaml {copy-lines="none",label="xnetwork"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+spec:
+  resources:
+    - name: vpc-resource
+      base:
+        apiVersion: ec2.aws.upbound.io/v1beta1
+        kind: VPC
+        # Removed for Brevity
+    - name: gateway-resource
+      base:
+        apiVersion: ec2.aws.upbound.io/v1beta1
+        kind: InternetGateway
+        # Removed for Brevity
+    - name: subnet-resource
+      base:
+        apiVersion: ec2.aws.upbound.io/v1beta1
+        kind: Subnet
+        # Removed for Brevity
+  compositeTypeRef:
+    apiVersion: aws.platformref.upbound.io/v1alpha1
+    kind: XNetwork
+```
+
+The {{<hover label="xnetwork" line="20">}}compositeTypeRef{{</hover>}} gives
+this Composition a 
+{{<hover label="xnetwork" line="21">}}apiVersion{{</hover>}} and
+{{<hover label="xnetwork" line="22">}}kind{{</hover>}} to reference in another
+Composition. 
+
+{{<hint "note" >}}
+The [Enabling a Composite Resource](#enabling-composite-resources) section 
+describes the 
+{{<hover label="xnetwork" line="20">}}compositeTypeRef{{</hover>}} field.
+{{< /hint >}}
+
+A second Composition, defining other resources, in this example, an AWS Elastic
+Kubernetes Cluster, can reference the previous
+{{<hover label="xnetwork" line="22">}}XNetwork{{</hover>}}
+
+```yaml {copy-lines="none",label="xcluster"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+spec:
+  resources:
+    - name: nested-network-composition
+      base:
+        apiVersion: aws.platformref.upbound.io/v1alpha1
+        kind: XNetwork
+        # Removed for brevity
+    - name: eks-cluster-resource
+      base:
+        apiVersion: eks.aws.upbound.io/v1beta1
+        kind: Cluster
+        # Removed for brevity
+```
+
+When a Composite Resource creates all the managed resources from this 
+Composition, the resources defined by the 
+{{<hover label="xcluster" line="8">}}XNetwork{{</hover>}} get created along with
+the EKS {{<hover label="xcluster" line="13">}}cluster{{</hover >}}. 
+
+{{<hint "note" >}}
+This abbreviated example is from the Upbound [AWS Reference
+Platform](https://github.com/upbound/platform-ref-aws).
+
+View the complete Compositions in the reference platform's 
+[package directory](https://github.com/upbound/platform-ref-aws/blob/main/package/cluster/composition.yaml).
+{{</hint >}}
+
+#### Cross-reference resources
+
+Inside a Composition some resources use identifiers or names of other resources.
+For example, creating a new `network` and applying the network identifier to a
+virtual machine. 
+
+Resources inside a Composition can cross-reference other resources by matching
+a label or a _controller reference_.
```

Comment:
Exposing the concept of controller references in this API is one of my many regrets. 😬